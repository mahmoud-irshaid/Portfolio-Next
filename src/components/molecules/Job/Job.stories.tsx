/* eslint-disable import/no-anonymous-default-export */
import { Job } from './Job';

import { mockJobs } from '@mockdata';
import type { Job as JobType } from '@types';

export default {
	title: 'molecules/Job',
	component: Job,
};

export const JobStory = (args: Omit<JobType, 'description'>) => (
	<Job {...args} />
);

JobStory.storyName = 'Job';
JobStory.args = {
	...mockJobs[0],
	children: <div>Some body</div>,
};
