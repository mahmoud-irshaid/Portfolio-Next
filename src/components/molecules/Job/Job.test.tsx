import { Job } from '../Job';

import { mockJobs } from '@mockdata';
import { render } from '@testing-library/react';

describe('Job', () => {
	it('renders correctly', () => {
		const { container } = render(<Job {...mockJobs[0]}>Some body</Job>);
		expect(container).toMatchSnapshot();
	});
});
