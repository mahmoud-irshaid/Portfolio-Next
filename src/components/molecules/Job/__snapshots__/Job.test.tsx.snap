// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job renders correctly 1`] = `
<div>
  <div
    class="mt-12 w-full glass-light dark:glass-dark mb-6 rounded-xl px-8 py-6 drop-shadow-lg"
  >
    <div
      class="-mt-12 mb-4 flex w-full justify-center"
    >
      <img
        alt="Holland & Barrett"
        class="rounded-md"
        data-nimg="1"
        decoding="async"
        height="64"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fmahmoudirshaid.com%2F_next%2Fimage%3Furl%3Dhttps%253A%252F%252Fcdn.sanity.io%252Fimages%252Fyq5nnpfd%252Fproduction%252F220cf610ee8a9bc6f821a5a3285ca7c224536cb8-128x128.jpg%26w%3D128%26q%3D75&w=128&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fmahmoudirshaid.com%2F_next%2Fimage%3Furl%3Dhttps%253A%252F%252Fcdn.sanity.io%252Fimages%252Fyq5nnpfd%252Fproduction%252F220cf610ee8a9bc6f821a5a3285ca7c224536cb8-128x128.jpg%26w%3D128%26q%3D75&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fmahmoudirshaid.com%2F_next%2Fimage%3Furl%3Dhttps%253A%252F%252Fcdn.sanity.io%252Fimages%252Fyq5nnpfd%252Fproduction%252F220cf610ee8a9bc6f821a5a3285ca7c224536cb8-128x128.jpg%26w%3D128%26q%3D75&w=128&q=75 2x"
        style="color: transparent;"
        width="64"
      />
    </div>
    <div
      class="flex items-start gap-4 md:items-center"
    >
      <div
        class="mb-2 w-full"
      >
        <h3
          class="text-center font-bold"
        >
          Senior Software Engineer
        </h3>
        <h4
          class="text-center"
        >
          <a
            class="underlined relative"
            href="https://www.hollandandbarrett.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Holland & Barrett
          </a>
        </h4>
        <p
          class="mb-4 text-center text-sm text-slate-500"
        >
          Aug 2020 â€” present
        </p>
        <p
          class="hidden text-sm md:block md:text-base"
        >
          <strong>
            Technologies: 
          </strong>
          TypeScript, React, Next.js, Redux, GraphQL, REST API, Jest, Cypress
        </p>
      </div>
    </div>
    <p
      class="text-sm md:hidden"
    >
      <strong>
        Technologies: 
      </strong>
      TypeScript, React, Next.js, Redux, GraphQL, REST API, Jest, Cypress
    </p>
    Some body
  </div>
</div>
`;
